lib/streamlit/elements/arrow.py

```diff
@@ -130,7 +145,39 @@ def _arrow_dataframe(
 
         proto.editing_mode = ArrowProto.EditingMode.READ_ONLY
 
-        marshall(proto, data, default_uuid)
+        if isinstance(data, pa.Table):
+            # For pyarrow tables, we can just serialize the table directly
+            proto.data = type_util.pyarrow_table_to_bytes(data)
+        else:
+            # For all other data formats, we need to convert them to a pandas.DataFrame
+            # thereby, we also apply some data specific configs
+
+            # Determine the input data format
+            data_format = type_util.determine_data_format(data)
+
+            if type_util.is_pandas_styler(data):
+                # If pandas.Styler uuid is not provided, a hash of the position
+                # of the element will be used. This will cause a rerender of the table
+                # when the position of the element is changed.
+                delta_path = self.dg._get_delta_path_str()
+                default_uuid = str(hash(delta_path))
+                # default_uuid is a string only if the data is a `Styler`,
+                # and `None` otherwise.
+                assert isinstance(
```

Comment:
Should this be an assert? Wouldn't that stop streamlit? Should we just be logging?